src/parser/loc.ml

```diff
@@ -28,3 +28,24 @@ let span spans =
 
 let nudge_start offset span =
   { span with start = { span.start with column = span.start.column + offset } }
+
+let to_loc span =
+  let loc_start =
+    Lexing.
+      {
+        pos_fname = span.file;
+        pos_lnum = span.start.line;
+        pos_bol = 0;
+        pos_cnum = span.start.column;
+      }
+  in
+  let loc_end =
+    Lexing.
+      {
+        pos_fname = span.file;
+        pos_lnum = span.end_.line;
+        pos_bol = 0;
+        pos_cnum = span.end_.column;
+      }
+  in
+  Location.{ loc_start; loc_end; loc_ghost = false }
```

Comment:
The intention here is a convenience for users of the library, though it needs a bit more work. When we first call `Odoc_parser.parse_comment` we pass in a `Lexing.position` to indicate the start position of the comment. The intent here is to be able to turn our internal `Loc.t` types back into the original type, more-or-less. The use of `Location.t` as a return type is because it conveniently represents a span between two `Lexing.position` values. While it does bring in a dependency on `compiler-libs`, I think it's pretty unlikely anyone using this library isn't _already_ using `compiler-libs`, though we could always return a pair of `Lexing.positions` instead. However, in order to do this correctly we need to stash the original `Lexing.position` away and keep it for this operation. I'll give this a try and see how it looks. 

As an example of where this function would be useful, see here: https://github.com/janestreet/ppx_js_style/blob/master/src/ppx_js_style.ml#L393-L432